include/sir/config.h

```diff
@@ -315,83 +402,115 @@
  *   Error in findneedle (haystack.c:384): 'Too much hay'
  *   ~~~
  */
-# define SIR_ERRORFORMAT "Error in %s (%s:%u): '%s'"
+# if !defined(SIR_ERRORFORMAT)
+#  define SIR_ERRORFORMAT "Error in %s (%s:%u): '%s'"
+# endif
 
 /** The string that represents any unknown. */
-# define SIR_UNKNOWN "<unknown>"
+# if !defined(SIR_UNKNOWN)
+#  define SIR_UNKNOWN "<unknown>"
+# endif
 
 /** stderr destination string. */
-# define SIR_DESTNAME_STDERR     "stderr"
+# if !defined(SIR_DESTNAME_STDERR)
+#  define SIR_DESTNAME_STDERR     "stderr"
+# endif
 
 /** stdout destination string. */
-# define SIR_DESTNAME_STDOUT     "stdout"
+# if !defined(SIR_DESTNAME_STDOUT)
+#  define SIR_DESTNAME_STDOUT     "stdout"
+# endif
 
 /** System logger destination string. */
-# define SIR_DESTNAME_SYSLOG     "syslog"
+# if !defined(SIR_DESTNAME_SYSLOG)
+#  define SIR_DESTNAME_SYSLOG     "syslog"
+# endif
 
 /** Fallback system logger identity. */
-# define SIR_FALLBACK_SYSLOG_ID  "libsir"
+# if !defined(SIR_FALLBACK_SYSLOG_ID)
+#  define SIR_FALLBACK_SYSLOG_ID  "libsir"
+# endif
 
 /** Fallback system logger category. */
-# define SIR_FALLBACK_SYSLOG_CAT "general"
+# if !defined(SIR_FALLBACK_SYSLOG_CAT)
+#  define SIR_FALLBACK_SYSLOG_CAT "general"
+# endif
 
 /**
  * The number of actual levels; ::SIRL_NONE, ::SIRL_ALL, and ::SIRL_DEFAULT
  * are pseudo levels and end up being mapped (or not) to the others.
  */
-# define SIR_NUMLEVELS 8
+# if !defined(SIR_NUMLEVELS)
```

Comment:
This is also fixed.

Reply:
K - I did these all with a regex replacement so :)